src/main/java/org/generationcp/middleware/api/germplasm/GermplasmServiceImpl.java

```diff
@@ -388,7 +388,11 @@ public long countAttributesByGUID(final String gemrplasmUUID, final List<String>
 		return this.daoFactory.getAttributeDAO().countAttributesByGUID(gemrplasmUUID, attributeDbIds);
 	}
 
-	private void saveGermplasmUpdateDTO(final Integer userId, final Map<String, Integer> attributeCodes,
+  @Override public Set<Integer> getGidsOfGermplasmWithDerivativeOrMaintenanceDescendants(final List<Integer> gids) {
+	return this.daoFactory.getGermplasmDao().getGidsOfGermplasmWithDerivativeOrMaintenanceDescendants(Sets.newHashSet(gids));
+  }
```

Comment:
You can merge these two methods, since there are basically the same: getGidsOfGermplasmWithDerivativeOrMaintenanceDescendants, getGidsOfDerivativeGermplasmWithDescendants